reference-implementation/test/writable-stream.js

```diff
@@ -1,25 +1,131 @@
 'use strict';
 const test = require('tape-catch');
 
-function writeArrayToStream(array, writableStream) {
-  array.forEach(chunk => writableStream.write(chunk));
-  return writableStream.close();
+function promise_rejects(t, expectedReason, promise, name, msg) {
+  promise.then(value => {
+    t.fail(name + ' fulfilled unexpectedly');
+    t.end();
+  }, reason => {
+    t.equal(reason, expectedReason, msg);
+  });
 }
 
-test('error argument is given to start method', t => {
-  let error;
+function writeArrayToStream(array, writableStreamWriter) {
+  array.forEach(chunk => writableStreamWriter.write(chunk));
+  return writableStreamWriter.close();
+}
+
+test('desiredSize on a released writer', t => {
+  const ws = new WritableStream({});
+  const writer = ws.getWriter();
+  writer.releaseLock();
+
+  try {
```

Comment:
You could use `t.throws(...)`? https://www.npmjs.com/package/tape#tthrowsfn-expected-msg